src/tests/unit/tests/background/stores/assessment-card-selection-store.test.ts

```diff
@@ -0,0 +1,392 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { AssessmentCardSelectionActions } from 'background/actions/assessment-card-selection-actions';
+import { AssessmentCardSelectionStore } from 'background/stores/assessment-card-selection-store';
+import { AssessmentCardSelectionStoreData } from 'common/types/store-data/assessment-card-selection-store-data';
+import { RuleExpandCollapseData } from 'common/types/store-data/card-selection-store-data';
+import { cloneDeep, forOwn } from 'lodash';
+
+import {
+    AssessmentCardSelectionPayload,
+    BaseActionPayload,
+    RuleExpandCollapsePayload,
+} from '../../../../../background/actions/action-payloads';
+import { StoreNames } from '../../../../../common/stores/store-names';
+import { createStoreWithNullParams, StoreTester } from '../../../common/store-tester';
+
+describe('AssessmentCardSelectionStore', () => {
+    test('constructor has no side effects', () => {
+        const testObject = createStoreWithNullParams(AssessmentCardSelectionStore);
+        expect(testObject).toBeDefined();
+    });
+
+    test('getId', () => {
+        const testObject = createStoreWithNullParams(AssessmentCardSelectionStore);
+
+        expect(testObject.getId()).toEqual(StoreNames[StoreNames.AssessmentCardSelectionStore]);
+    });
+
+    test('check defaultState is as expected', () => {
+        const defaultState = getDefaultState();
+
+        expect(defaultState.testKey.rules).toBeNull();
+    });
+
+    function getDefaultState(): AssessmentCardSelectionStoreData {
+        return createStoreWithNullParams(AssessmentCardSelectionStore).getDefaultState();
+    }
+});
+
+describe('AssessmentCardSelectionStore Test', () => {
+    let initialState: AssessmentCardSelectionStoreData = null;
+    let expectedState: AssessmentCardSelectionStoreData = null;
+
+    beforeEach(() => {
+        const defaultState: AssessmentCardSelectionStoreData = {
+            testKey: {
```

Comment:
Might make sense to use a test default state with multiple test keys to ensure that the logic will work in the future. So instead of 
```
testKey: {....testData}
```
this might be
```
'test-key-1': {....testData1},
'test-key-2': {....testData2}
```